controller/src/beerocks/master/wfa_ca.cpp

```diff
@@ -0,0 +1,33 @@
+#include "wfa_ca.h"
+
+#include <beerocks/bcl/beerocks_string_utils.h>
+#include <beerocks/bcl/network/network_utils.h>
+
+#include <beerocks/tlvf/beerocks_message.h>
+#include <beerocks/tlvf/beerocks_message_bml.h>
+
+#include <easylogging++.h>
+
+using namespace beerocks;
+
+void wfa_ca::handle_wfa_ca_message(
+    Socket *sd, std::shared_ptr<beerocks_message::cACTION_HEADER> beerocks_header,
+    ieee1905_1::CmduMessageRx &cmdu_rx, ieee1905_1::CmduMessageTx &cmdu_tx, db &database,
+    task_pool &tasks)
+{
+    auto request = cmdu_rx.addClass<beerocks_message::cACTION_BML_WFA_CA_CONTROLLER_REQUEST>();
+
+    if (!request) {
+        LOG(ERROR) << "addClass cACTION_BML_WFA_CA_CONTROLLER_REQUEST failed";
+        return;
+    }
+
+    auto command_tuple = request->command(0);
+    if (!std::get<0>(command_tuple)) {
+        LOG(ERROR) << "getting command entry has failed!";
+        return;
+    }
+    const char *command_ptr = (char *)(&std::get<1>(command_tuple));
```

Comment:
Please, no C-style casts.

Reply:
This piece of code has changed in the follow-up commit 998fe23e9e17a6a5c3653bfb522920d93526d8ed.
BTW c-like implicit casting notation is "( new_type ) expression".
CPP implict casting notation is "new_type ( expression )".
Since "char*(expression)" I put "char*" inside parantesis.

Reply:
`new_type (expression)` only works if `new_type` is a type name, not if it is a declarator. And it doesn't replace casts, it replaces conversions. `(int) expr` is a conversion to int, so it really is equivalent to `int(expr)`. For the `char*` case, however, even with the parenthesis, this is still parsed as a C-style cast.

More importantly, though, the programmer parses it as a C-style cast.

So, we should make this explicit (IMO) and call it a reinterpret_cast, because that's what it is.